BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/XFintoPUEO.py

```diff
@@ -0,0 +1,345 @@
+            for q in range(2):
+                PUEO_v0.write(str(OutputVV[p][q])+" ")
+            PUEO_v0.write("\n")
+        print("Done with vv_0 for Antenna " +str(antenna+1))
+
+# For hh_0
+for antenna in range(g.NPOP):
+    with open(f'{g.WorkingDir}/Test_Outputs/'+"hh_0_"+{g.gen}+" "+str(antenna+1), "w+") as PUEO_h0:
+        os.chmod(f'{g.WorkingDir}/Test_Outputs/'+"hh_0_"+{g.gen}+" "+str(antenna+1), 0o777)
+
+        uanMaxG = []
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) #Gives Gain Column           
+            MaxGain = uanDatG[1314] #Gain at theta = 90                        
+            uanMaxG.append(MaxGain)
+
+        OutputHH = np.column_stack((freqArray, uanMaxG))
+        for p in range(numFreq):
+            for q in range(2):
+                PUEO_h0.write(str(OutputHH[p][q])+" ")
+            PUEO_h0.write("\n")
+        print("Done with hh_0 for Antenna " +str(antenna+1))
+
+# For vh_0: 
+for antenna in range(g.NPOP):
+    with open(f'{g.WorkingDir}/Test_Outputs/'+"vh_0_"+{g.gen}+" "+str(antenna+1), "w+") as PUEO_vh0:
+        os.chmod(f'{g.WorkingDir}/Test_Outputs/'+"vh_0_"+{g.gen}+" "+str(antenna+1), 0o777)
+        uanMaxG = []
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) #Gives Gain Column           
+            MaxGain = uanDatG[1314] #Gain at theta = 90                        
+            uanMaxG.append(MaxGain)
+
+        OutputVH = np.column_stack((freqArray, uanMaxG))
+        for p in range(numFreq):
+            for q in range(2):
+                PUEO_vh0.write(str(OutputVH[p][q])+" ")
+            PUEO_vh0.write("\n")
+        print("Done with vh_0 for Antenna " +str(antenna+1))
+
+# for hv_0:
+for antenna in range(g.NPOP):
+    with open(f'{g.WorkingDir}/Test_Outputs/'+"hv_0_"+{g.gen}+" "+str(antenna+1), "w+") as PUEO_hv0:
+        os.chmod(f'{g.WorkingDir}/Test_Outputs/'+"hv_0_"+{g.gen}+" "+str(antenna+1), 0o777)
+        uanMaxG = []
+        for freq in range(numFreq):
+            uanDatG = readFile(antenna+1, freq+1) #Gives Gain Column           
+            MaxGain = uanDatG[1314] #Gain at theta = 90                        
+            uanMaxG.append(MaxGain)
+
+        OutputHV = np.column_stack((freqArray, uanMaxG))
+        for p in range(numFreq):
+            for q in range(2):
+                PUEO_hv0.write(str(OutputHV[p][q])+" ")
+            PUEO_hv0.write("\n")
+        print("Done with hv_0 for Antenna " +str(antenna+1))
+
+# Now to pull angles 5, 10, 20, 30, 45, 90 for el or az and respective theta or phi
+# Angles and index locations. For theta. 5:[73], 10:[146], 20:[292], 30:[438], 45:[657], 90:[1314]
+# For phi. 5:[1], 10:[2], 20:[4], 30:[6], 45:[9], 90:[18]
+
+# Creating new Array with Frequencies 6 times
+BIGfreqArray = []
+for z in range(6):
+    for q in range(numFreq):
+        scinoteval = "{:.2e}".format(freqVals[q])
+        BIGfreqArray.append(scinoteval)
```

Comment:
As above:
```suggestion
big_freq_list = [f'{val:.2e}' for val in freqVals] * 6
```
or, if they have different len:
```suggestion
big_freq_list = [f'{freqVals[q]:.2e}' for q in range(numFreq)] * 6
```
Note that this also fixes the var name, so if you apply this suggestion you will also need to update the name below.